### PR TITLE
Dx: Convert JSX to TSX

### LINKED ISSUE #181: [DX] Convert components to tsx
### Description

Recently merged #179 introduced some components written in jsx. While that is not bad, we could transform them to tsx to align with the project stack and also improves future developments on those files.

If needed, I could work on that :)

### PR DESCRIPTION
**Pull request by CerealPlayer**
### 🛠️ Fixes Issue

Closes #181 

### ➕ Changes Introduced
Converted al jsx files to tsx

### 📄 Note To Reviewers

Types are not too good. As per conversation with @ykdojo over discord, we just want things to move fast and work.

### CODE REVIEW

**Review by ykdojo — APPROVED** [review]

File: src/components/CallToAction.tsx [hunk 1]
```diff
@@ -2,14 +2,13 @@ import Image from 'next/future/image';
...
 import { Container } from '@/components/Container';
-import backgroundImage from '@/images/background-call-to-action.jpg';
 
 export function CallToAction() {
 	return (
 		<section id='join-conversation' className='relative overflow-hidden bg-blue-600 py-32'>
 			<Image
 				className='absolute top-1/2 left-1/2 max-w-none -translate-x-1/2 -translate-y-1/2'
-				src={backgroundImage}
+				src='../images/background/background-call-to-action.jpg'
```
[the remark this answers is not in the document]
Comment (CerealPlayer) [reply src/components/CallToAction.tsx:11]: For this one in particular Typescript compiler complained about the previous approach. Using this path was the next better thing that worked

File: src/components/Features.tsx [hunk 2]
```diff
@@ -1,15 +1,22 @@
-import { useId } from 'react';
+import React, { useId } from 'react';
```
[the remark this answers is not in the document]
Comment (CerealPlayer) [reply src/components/Features.tsx:1]: You are right, I just went for the fastest thing I could do given we just want to launch, I'll make the changes

**Comment by ykdojo** [comment]
@CerealPlayer are you still working on it? Looks like we have conflicts now

**Comment by CerealPlayer** [comment]
@ykdojo Yeah I saw, however seems like these conflicts need write permissions in order to be merged.

Anyway, I see there are new jsx components so I could make a few more commits

**Comment by CerealPlayer** [comment]
Just fixed conflicts and converted all jsx elements to tsx elements @ykdojo

**Review by ykdojo — APPROVED** [review]

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -1,21 +1,22 @@
-import { Fragment } from 'react';
+import React, { Fragment } from 'react';
```
Comment (subhoghoshX) [inline src/components/Header/Header.tsx:1]:
```suggestion
import  { ReactNode, Fragment } from 'react';
```

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -1,21 +1,22 @@
...
 import { Popover, Transition } from '@headlessui/react';
 import clsx from 'clsx';
 
 import { Button } from '@/components/Button';
 import { Container } from '@/components/Container';
 import { NavLink } from '@/components/NavLink';
+import { Session } from '@supabase/supabase-js';
 
-function MobileNavLink({ href, children }) {
+function MobileNavLink({ href, children }: { href: string; children: React.ReactNode }) {
```
Comment (subhoghoshX) [inline src/components/Header/Header.tsx:11]:
```suggestion
function MobileNavLink({ href, children }: { href: string; children: ReactNode }) {
```

File: src/components/NavLink.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import Link from 'next/link';
+import React from 'react';
```
Comment (subhoghoshX) [inline src/components/NavLink.tsx:2]:
```suggestion
import { ReactNode } from 'react';
```

File: src/components/NavLink.tsx [hunk 6]
```diff
@@ -1,6 +1,7 @@
...
 
-export function NavLink({ href, children }) {
+export function NavLink({ href, children }: { href: string; children: React.ReactNode }) {
```
Comment (subhoghoshX) [inline src/components/NavLink.tsx:4]:
```suggestion
export function NavLink({ href, children }: { href: string; children: ReactNode }) {
```

**Review by subhoghoshX — APPROVED** [review]

**Comment by CerealPlayer** [comment]
@subhoghoshX Thank you! And thanks Github for these commit suggestiong features 😍

**Review by ykdojo — APPROVED** [review]

**Comment by subhoghoshX** [comment]
> @subhoghoshX Thank you! And thanks Github for these commit suggestions features :heart_eyes:

Yup, that feature is awesome. Btw I can see some Prettier and TypeScript checks failing, could you look into that? Specifically, it can't seem to find this image `@/images/background-call-to-action.jpg`.

**Review by subhoghoshX — APPROVED** [review]

**Comment by CerealPlayer** [comment]
Thanks @subhoghoshX . I had to move images to public folder (as far as I know that is the recommended approach NextJS) and point to relative path in Image component as typescript was complaining. Locally it works, but for some reason the GH actions will not allow it.

**Pull request merged**